### PR TITLE
Add check to ensure Glean is only initialized on the main process.

### PR DESCRIPTION
**Pull request by travis79**
Trying to initialize Glean on a process other than the main process results in an error when Glean tries to use WorkManager because WorkManager is only initialized on the main process.  This checks whether initialize is being called by the main process and returns if it isn't.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt [hunk 1]
```diff
@@ -88,6 +89,13 @@ open class GleanInternalAPI internal constructor () {
         applicationContext: Context,
         configuration: Configuration = Configuration()
     ) {
+        // In certain situations Glean.initialize may be called from a process other than the main
+        // process.  In this case we want initialize to be a no-op and just return.
+        if (!applicationContext.isMainProcess()) {
+            logger.error("Attempted to initialize Glean on a process other than the main process")
```
Comment (Dexterp37) [inline components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt]: @travis79 can you make sure that it's ok to call the `Logger` on other processes?

Comment (travis79) [reply components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt]: After experimenting with this, it appears to log as intended.

**Comment by dector** [comment]
Just wondering if runtime exception should be thrown as well?
If Glean component is required to be initialized in main thread - doing this in another thread seems like configuration error. :thinking:

**Comment by dector** [comment]
Awesome, thanks!

**Comment by travis79** [comment]
@dector I've filed an investigation bug [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1558181) in case you are interested in following it.  It's sort of a broad bug for now to cover a full investigation on the pros and cons and other pitfalls of using Glean in a multi-process application.  It will likely result in more specific tasks as we dig into it.

**Comment by dector** [comment]
@travis79 Great, thank you! I've created follow-up issue in AC project: #3342 .

**Pull request merged**